### ISSUE TITLE
With Computed fields on multiple owned types, sql attempts to update those fields

### ISSUE THREAD

**Issue by lakeman** [dotnet/efcore#24441]
Define one owned object with a computed field, and everything works fine.

Define a second instance of the same owned type on the same parent, and saving will attempt to update the computed fields.(#19176 might be related).

Repro;
```C#
    public class OwnedType
    {
        public string Computed { get; set; }
        public string Field { get; set; }
    }
    public class ParentType
    {
        public int Id { get; set; }
        public OwnedType First { get; set; }
        public OwnedType Second { get; set; }
    }

    public class Context : DbContext
    {
        protected override void OnConfiguring(DbContextOptionsBuilder optionsBuilder)
        {
            optionsBuilder.UseSqlServer("Server=localhost;Database=Repro;integrated security=True;Trusted_Connection=true;");
            base.OnConfiguring(optionsBuilder);
        }

        protected override void OnModelCreating(ModelBuilder modelBuilder)
        {
            modelBuilder.Entity<ParentType>(p =>
            {
                p.OwnsOne(p => p.First, o =>
                {
                    o.Property(x => x.Field)
                        .HasColumnName("FirstField");
                    o.Property(x => x.Computed)
                        .HasColumnName("FirstComputed")
                        .HasComputedColumnSql("FirstField+'x'", false);
                });

                if (Program.BROKEN)
                {
                    p.OwnsOne(p => p.Second, o =>
                    {
                        o.Property(x => x.Field)
                            .HasColumnName("SecondField");
                        o.Property(x => x.Computed)
                            .HasColumnName("SecondComputed")
                            .HasComputedColumnSql("SecondField+'x'", false);
                    });
                }
                else
                {
                    p.Ignore(p => p.Second);
                }
            });

            base.OnModelCreating(modelBuilder);
        }
    }

    class Program
    {
        public const bool BROKEN = true;
        static async Task Main(string[] args)
        {
            using var db = new Context();
            await db.Database.EnsureDeletedAsync();
            await db.Database.EnsureCreatedAsync();

            var obj = new ParentType
            {
                First = new OwnedType{ Field = "Value1" }
            };
            if (BROKEN)
                obj.Second = new OwnedType{ Field = "Value2" };

            db.Set<ParentType>().Add(obj);
            await db.SaveChangesAsync();
        }
    }
```

If I look at the model's debug string;
```c#
db.Model.ToDebugString(MetadataDebugStringOptions.LongDefault)
```
I can see an obvious difference between the two use cases;
```
Working;
  EntityType: OwnedType
      Computed (string) BeforeSave:Ignore AfterSave:Ignore ValueGenerated.OnAddOrUpdate
Broken;
  EntityType: ParentType.First#OwnedType
      Computed (string)
```

It looks like the default fluent configuration isn't working for this case.

**Comment by lakeman** [comment]
Fiddling with the fluent API a bit more, I was surprised that this;
```c#
var b1 = p.OwnsOne(p => p.First);
var b2 = p.OwnsOne(p => p.Second);

b1.Property(x => x.Field)
    .HasColumnName("FirstField");
```

Triggers a NullReferenceException;
```
  StackTrace:
   at Microsoft.EntityFrameworkCore.Metadata.Builders.OwnedNavigationBuilder`2.<>c__DisplayClass4_0`1.<Property>b__0()
   at Microsoft.EntityFrameworkCore.Metadata.Conventions.Internal.ConventionDispatcher.Run[T](Func`1 func, IConventionForeignKey& foreignKey)
   at Microsoft.EntityFrameworkCore.Metadata.Builders.OwnedNavigationBuilder.UpdateBuilder[T](Func`1 configure)
   at Microsoft.EntityFrameworkCore.Metadata.Builders.OwnedNavigationBuilder`2.Property[TProperty](Expression`1 propertyExpression)
   at Repro.Context.<>c.<OnModelCreating>b__1_0(EntityTypeBuilder`1 p) in ...
```

**Comment by AndriySvyryd** [comment]
The first issue is fixed in d788e06ee73b83674099325d1988469f4f6ac720
The second one in 9fa54fec3a80377d6c1c5c2fa227bd04cde2ebe4
